packages/cli/test/integration/publicApi/tags.test.ts

```diff
@@ -0,0 +1,338 @@
+import type { SuperAgentTest } from 'supertest';
+import Container from 'typedi';
+import type { User } from '@db/entities/User';
+import { TagRepository } from '@db/repositories/tag.repository';
+
+import { randomApiKey } from '../shared/random';
+import * as utils from '../shared/utils/';
+import * as testDb from '../shared/testDb';
+import { createUser } from '../shared/db/users';
+import { createTag } from '../shared/db/tags';
+
+let owner: User;
+let member: User;
+let authOwnerAgent: SuperAgentTest;
+let authMemberAgent: SuperAgentTest;
+
+const testServer = utils.setupTestServer({ endpointGroups: ['publicApi'] });
+
+beforeAll(async () => {
+	owner = await createUser({
+		role: 'global:owner',
+		apiKey: randomApiKey(),
+	});
+
+	member = await createUser({
+		role: 'global:member',
+		apiKey: randomApiKey(),
+	});
+});
+
+beforeEach(async () => {
+	await testDb.truncate(['Tag']);
+
+	authOwnerAgent = testServer.publicApiAgentFor(owner);
+	authMemberAgent = testServer.publicApiAgentFor(member);
+});
+
+const testWithAPIKey =
+	(method: 'get' | 'post' | 'put' | 'delete', url: string, apiKey: string | null) => async () => {
+		void authOwnerAgent.set({ 'X-N8N-API-KEY': apiKey });
+		const response = await authOwnerAgent[method](url);
+		expect(response.statusCode).toBe(401);
+	};
+
+describe('GET /tags', () => {
+	test('should fail due to missing API Key', testWithAPIKey('get', '/tags', null));
+
+	test('should fail due to invalid API Key', testWithAPIKey('get', '/tags', 'abcXYZ'));
```

Comment:
Ideally let's try to avoid being so dry with tests - this makes them easier to read and maintain. If need be we can always use `.each` for table-style tests.

Reply:
Since other parts of the public API follow the same process, this approach to me looks even better than what we currently have, since it avoids code duplication